src/applications/vaos/appointment-list/redux/selectors.js

```diff
@@ -374,10 +374,14 @@ export function selectExpressCareAvailability(state) {
   };
 }
 
-export function selectExpressCareRequestById(state, id) {
-  const { appointmentDetails, pending, past, confirmed } = state.appointments;
+export function selectAppointmentById(state, id, types = null) {
```

Comment:
I collapse the selectors for single appointment/requests down into one, and am passing the types to try address any possible id collisions.